src/persist-client/src/internal/paths.rs

```diff
@@ -49,6 +51,60 @@ impl PartId {
     }
 }
 
+/// A component that provides information about the writer of a blob.
+/// For older blobs, this is a UUID for the specific writer instance;
+/// for newer blobs, this is a string representing the version at which the blob was written.
+/// In either case, it's used to help determine whether a blob may eventually
+/// be linked into state, or whether it's junk that we can clean up.
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Clone)]
+pub enum WriterKey {
+    Id(WriterId),
+    Version(String),
+}
+
+impl WriterKey {
+    /// This writes a version of the format 0-BBBB.CCCC, where BBBB is the major
+    /// version and CCCC is the minor version. (We include the 0 so we can
+    /// change versioning schemes in the future without burning another single-letter
+    /// prefix.)
+    pub fn for_version(version: &Version) -> WriterKey {
+        assert!(version.major <= 9999);
+        assert!(version.minor <= 9999);
```

Comment:
Using a `.min(9999)` here would avoid the panics and still pass tests, in case you prefer it.